service/controller/v5/resource/blobobject/blob_client.go

```diff
@@ -0,0 +1,119 @@
+package blobobject
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"net/url"
+	"strings"
+
+	"github.com/giantswarm/microerror"
+
+	"github.com/Azure/azure-sdk-for-go/services/storage/mgmt/2018-07-01/storage"
+	"github.com/Azure/azure-storage-blob-go/azblob"
+)
+
+const (
+	blobFormatString      = `https://%s.blob.core.windows.net`
+	resourceNotFoundError = "NotFound"
+	maxRetriesRequests    = 3
+)
+
+type BlobClient struct {
+	storageAccountsClient *storage.AccountsClient
+	containerURL          azblob.ContainerURL
+}
+
+func (c *BlobClient) BlobExists(ctx context.Context, blobName string) (bool, error) {
+	blobURL := c.containerURL.NewBlockBlobURL(blobName)
+
+	_, err := blobURL.GetProperties(ctx, azblob.BlobAccessConditions{})
+	if IsBlobNotFound(err) {
+		return false, nil
+	}
+	if err != nil {
+		return false, microerror.Mask(err)
+	}
+
+	return true, nil
+}
+
+func (c *BlobClient) createBlockBlob(ctx context.Context, blobName string, payload string) (azblob.BlockBlobURL, error) {
```

Comment:
This is used inside this package. But let's please think as it was used outside. We should have exported methods which are used outside this object and private methods which are used only internally. If you are afraid this object can be used outside package which I think it's legit just make it private by renaming it to `blobClient`.

Reply:
> But let's please think as it was used outside

But why? Also I don't want to make `BlobClient` private as it will be used within `instance` resource. How we can do that in a different way?

Reply:
Then put it in a different package. We don't have dependencies between resources. That would lead to a spaghetti code and circular dependencies. My proposition is `v5/azureclient/`.

Reply:
I could do that in that way. But every `BlobClient` is related to the specific blobObject as it operates over blob objec with `ContainerURL`. If that could be possible to a setup unique client for all objects - I would implement that in the `azure-operator/clients` package

Reply:
Hm, I took a look into code again and the move to the `v5/blobclient` might work